pkg/helmcli/repository/adder.go

```diff
@@ -0,0 +1,148 @@
+package repository
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gojekfarm/albatross/pkg/helmcli/flags"
+	"github.com/gojekfarm/albatross/pkg/logger"
+
+	"github.com/gofrs/flock"
+	"gopkg.in/yaml.v2"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+const timeout time.Duration = 30 * time.Second
+
+type adder struct {
+	flags.AddFlags
+	settings *cli.EnvSettings
+}
+
+func (o *adder) Add(ctx context.Context) (*repo.Entry, error) {
+	err := o.checkPrerequisite()
+	if err != nil {
+		return nil, err
+	}
+
+	// Acquire a file lock for process synchronization
+	fileLock := flock.New(strings.Replace(o.RepoFile, filepath.Ext(o.RepoFile), ".lock", 1))
+	lockCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	locked, err := fileLock.TryLockContext(lockCtx, time.Second)
+	if err == nil && locked {
```

Comment:
Is there a case where err is nil and locked is false? Would a single check suffice here? If yes, the next if can be else.

Reply:
https://github.com/gofrs/flock/blob/master/flock_unix.go#L154